WebDriverAgentLib/Categories/XCAXClient_iOS+FBSnapshotReqParams.m

```diff
@@ -33,15 +45,34 @@ id FBGetCustomParameterForElementSnapshot (NSString *name)
   return customRequestParameters[name];
 }
 
-static id swizzleDefaultParameters(id self, SEL _cmd)
+static id swizzledDefaultParameters(id self, SEL _cmd)
 {
-  NSMutableDictionary *result = [NSMutableDictionary dictionaryWithDictionary:original_defaultParameters(self, _cmd)];
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    NSMutableDictionary *params = [NSMutableDictionary dictionaryWithDictionary:original_defaultParameters(self, _cmd)];
+    [params addEntriesFromDictionary:defaultAdditionalRequestParameters];
+    defaultRequestParameters = params.copy;
+  });
+  if (nil == defaultRequestParameters) {
```

Comment:
Is this nil check needed given it's populated first in the first call and not cleared?

Reply:
just a precaution in case this method is called by `snapshotParameters`

Reply:
In that case, won't it be a problem that `defaultAdditionalRequestParameters` would be nil and won't be populated again since it's in a `dispatch_once` block?